staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/streaming.go

```diff
@@ -0,0 +1,249 @@
+	if ok {
+		return true, streamingEncodeUnstructuredList(w, list)
+	}
+	return false, nil
+}
+
+func getListMeta(list runtime.Object) (metav1.TypeMeta, metav1.ListMeta, []runtime.Object, error) {
+	v, err := conversion.EnforcePtr(list)
+	if err != nil {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, err
+	}
+	listType := v.Type()
+	if listType.NumField() != 3 {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected ListType to have 3 fields")
+	}
+	listMetaType, ok := listType.FieldByName("ListMeta")
+	if !ok {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected ListMeta field")
+	}
+	if listMetaType.Tag.Get("json") != "metadata,omitempty" {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected ListMeta field tag")
+	}
+	listMetaValue := v.FieldByName("ListMeta")
+	if !listMetaValue.IsValid() {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected ListMeta field")
+	}
+	listMeta, ok := listMetaValue.Interface().(metav1.ListMeta)
+	if !ok {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected ListMeta field to have ListMeta type")
+	}
+	typeMetaField, ok := listType.FieldByName("TypeMeta")
+	if !ok {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field")
+	}
+	if typeMetaField.Tag.Get("json") != ",inline" {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field tag")
+	}
+	typeMetaValue := v.FieldByName("TypeMeta")
+	if !typeMetaValue.IsValid() {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field")
+	}
+	typeMeta, ok := typeMetaValue.Interface().(metav1.TypeMeta)
+	if !ok {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field to have TypeMeta type")
+	}
+
+	itemsField, ok := listType.FieldByName("Items")
+	if !ok {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected Items field")
+	}
+	if itemsField.Tag.Get("json") != "items" {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field tag")
+	}
+	items, err := meta.ExtractList(list)
+	if err != nil {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, err
+	}
+	return typeMeta, listMeta, items, nil
+}
+
+func streamingEncodeList(w io.Writer, typeMeta metav1.TypeMeta, listMeta metav1.ListMeta, items []runtime.Object) error {
+	_, err := w.Write([]byte(`{`))
+	if err != nil {
+		return err
+	}
+	if typeMeta.Kind != "" {
+		err := encodeValue(w, []byte(`"kind":`), typeMeta.Kind, []byte(","))
+		if err != nil {
+			return err
+		}
+	}
+	if typeMeta.APIVersion != "" {
+		err := encodeValue(w, []byte(`"apiVersion":`), typeMeta.APIVersion, []byte(","))
+		if err != nil {
+			return err
+		}
+	}
+	err = encodeValue(w, []byte(`"metadata":`), listMeta, []byte(`,"items":`))
+	if err != nil {
+		return err
+	}
+	if items == nil {
+		_, err = w.Write([]byte("null}\n"))
```

Comment:
it's hard to follow the final `}\n` that is written to be sure the overall written object is correct (it is getting written in three different places where we hit return branches)

Reply:
Done